src/torrent.cpp

```diff
@@ -8252,8 +8252,9 @@ namespace libtorrent
 		}
 		m_became_finished = clamped_subtract_u16(m_became_finished, seconds);
 
-		m_last_upload = clamped_subtract_s16(m_last_upload, seconds);
-		m_last_download = clamped_subtract_s16(m_last_download, seconds);
+		// don't care about overflow because it would take 68 years
+		m_last_upload = m_last_upload - seconds;
+		m_last_download = m_last_download - seconds;
```

Comment:
I wonder if all relative timestamps that are incremented every second should be replaced by absolute timetstamps. The place where the space savings are the most relevant are probably in the peer list, so it might make sense to keep it there.
another option would be to decrease the precision to 2 seconds (to double the range)